### PR TITLE
Validate method for oci/ocm repository specs

### PR DESCRIPTION
**Pull request by mandelsoft**
#### What this PR does / why we need it:

This PR introduces a `Validate` method for OCI and OCM repository specifications.

Its task is to validate the spec as well as the connectivity (and potentially the credentials), if possible.
Checking credentials makes only sense, if there is no possibility to use different credentials for different shards
of the repository (for example organizations in the GitHub OCI registry).

### CODE REVIEW

File: api/oci/extensions/repositories/ocireg/type.go [hunk 1]
```diff
@@ -128,6 +130,18 @@ func (a *RepositorySpec) Repository(ctx cpi.Context, creds credentials.Credentia
...
 
+func (a *RepositorySpec) Validate(ctx cpi.Context, creds credentials.Credentials, context ...credentials.UsageContext) error {
+	info, err := a.getInfo(creds)
+	if err != nil {
+		return err
+	}
+	h, p, _ := info.HostInfo()
+	if p == "" {
+		p = "443"
+	}
```
Comment (hilmarf) [inline api/oci/extensions/repositories/ocireg/type.go:141]: should we log here something like:
'No port provided falling back to default port 443'

File: api/oci/extensions/repositories/docker/docker_test.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+//go:build docker_test
```
Comment (hilmarf) [inline api/oci/extensions/repositories/docker/docker_test.go:1]: when and where is this test executed?

**Review by hilmarf — CHANGES_REQUESTED** [review]
can we please have at least one place where the new 'Validate()' is go-documented?

File: api/oci/extensions/repositories/ocireg/type.go (hunk 1 above)
Comment (mandelsoft) [reply api/oci/extensions/repositories/ocireg/type.go:141]: this is the default port used for OCI registry REST access. I think a log is not required, here

File: api/oci/extensions/repositories/docker/docker_test.go (hunk 2 above)
Comment (mandelsoft) [reply api/oci/extensions/repositories/docker/docker_test.go:1]: you can execute it locally. Docker is not necessarily available when executing the standard tests.
The same has been done for the redis pubsub extension.

Comment (hilmarf) [reply api/oci/extensions/repositories/docker/docker_test.go:1]: okay, do we have an overview which additional integration tests are available?
are those already executed in https://github.com/open-component-model/ocm-integrationtest/ ?

**Review by hilmarf — APPROVED** [review]

**Review by hilmarf — APPROVED** [review]

**Pull request merged**